oak/server/rust/oak_config_serializer/src/main.rs

```diff
@@ -0,0 +1,127 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+//! An utility for serializing an Oak application configuration.
+//!
+//! Invoke with:
+//!
+//! ```shell
+//! cargo run --manifest-path=oak/server/rust/oak_config_serializer/Cargo.toml -- \
+//!     --config-file=<CONFIGURATION_FILE> \
+//!     --output-file=<OUTPUT_FILE>
+//! ```
+
+use anyhow::Context;
+use log::debug;
+use oak_abi::proto::oak::application::{
+    node_configuration::ConfigType, ApplicationConfiguration, NodeConfiguration,
+    WebAssemblyConfiguration,
+};
+use oak_utils::io::{read_file, write_file};
+use prost::Message;
+use std::{collections::HashMap, fs::read_to_string};
+use structopt::StructOpt;
+
+/// Command line options for the Oak Application Configuration Serializer.
+#[derive(StructOpt, Clone, Debug)]
+#[structopt(about = "Oak Application Configuration Serializer")]
+struct Opt {
+    #[structopt(long, help = "TOML application configuration file")]
+    config_file: String,
+    #[structopt(long, help = "Output file")]
+    output_file: String,
+}
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(deny_unknown_fields)]
+struct Config {
+    name: String,
+    modules: HashMap<String, Module>,
+    initial_node_configuration: InitialNodeConfig,
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+struct Module {
+    url: Option<String>,
+    path: Option<String>,
+}
+
+#[derive(serde::Deserialize, Debug)]
+#[serde(deny_unknown_fields)]
+struct InitialNodeConfig {
+    wasm_module_name: String,
+    wasm_entrypoint_name: String,
+}
+
+/// Load Wasm module from URL or a file, if URL is not specified.
+fn load_module(module: &Module) -> anyhow::Result<Vec<u8>> {
+    match &module.url {
+        Some(_url) => unimplemented!(),
+        None => {
+            let path = module.path.as_ref().context("Path is not specified")?;
+            read_file(&path)
+        }
+    }
+}
+
+/// Serializes an application configuration from `app_config` and writes it into `filename`.
+pub fn write_config_to_file(
+    app_config: &ApplicationConfiguration,
+    filename: &str,
+) -> anyhow::Result<()> {
+    let mut bytes = Vec::new();
+    app_config
+        .encode(&mut bytes)
+        .context("Couldn't encode application configuration")?;
+
+    write_file(&bytes, filename)?;
+    Ok(())
+}
+
+fn main() -> anyhow::Result<()> {
+    env_logger::init();
+
+    let opt = Opt::from_args();
+    debug!("parsed opts: {:?}", opt);
+
+    let config_file = read_to_string(opt.config_file)?;
+    let config: Config = toml::from_str(&config_file).context("Couldn't parse TOML config file")?;
+    debug!("parsed config file: {:?}", config);
+
+    // Load Wasm modules.
+    let mut modules = HashMap::new();
+    for (name, module) in config.modules.iter() {
+        modules.insert(name.to_string(), load_module(&module)?);
```

Comment:
How about something like


```suggestion
        modules.insert(name.to_string(), load_module(&module).context("could not load module from file {}", module)?);
```